pallets/settlement/src/lib.rs

```diff
@@ -112,6 +115,8 @@ pub trait Config:
     type MaxLegsInInstruction: Get<u32>;
     /// Weight information for extrinsic of the settlement pallet.
     type WeightInfo: WeightInfo;
+    /// Maximum number of assets that can be transferred in one leg.
+    type MaxNumberOfLegAssets: Get<u8>;
```

Comment:
Is this checked?

Reply:
This was the old constant used when we were thinking about accepting multiple fungible assets. Now, I changed the constant to check for the maximum number of NFTs being transferred, and it is used inside the `ensure_valid_legs` method.